### DISCUSSION TITLE
Building on Apple Silicon

### DISCUSSION THREAD

**Discussion by richardssam** [AcademySoftwareFoundation/OpenRV#487] [Build Issues]
Hi, I was able to run the current build on osx with xcode-15, but realized that its only building on x86_64 and not the ARM build.

I started going through seeing how complicated it might be to switch to include ARM, and am currently stuck on PyOpenGL_accelerate not wanting to build. This seems like a pretty dead package, how important is it to RV?

**Comment by bernie-laberge** [comment]
Hello @richardssam,
The ARM build was merged 3 weeks ago and then quickly reverted while investigating an issue to make sure no Open RV user would be blocked by it.
The ARM build commit is the following if you are interested: https://github.com/AcademySoftwareFoundation/OpenRV/commit/e78d986e9333a0ef05787b91e53afb7544fa39d1

But I don't remember any issue about the python requirement PyOpenGL_accelerate not wanting to build...
In any case, I believe it is only used by the optional RV package "EXR Data and Display Window Indicators".
So I don't think it will affect the rest of RV if you comment it out for now in your tests.

**Reply by richardssam** [reply]
Thanks so much, I'll give it a go.

**Comment by richardssam** [comment]
Is it possible to post the recipe for building Qt-5.15 on apple ARM hardware? I'm struggling to build it and did notice that vcpkg has a number of patches, which I'm assuming may be needed?

**Comment by cedrik-fuoco-adsk** [comment]
The documentation for MacOS has been updated today. You can take a look here https://aswf-openrv.readthedocs.io/en/latest/build_system/config_macos.html#install-qt. I've added the steps I did to compile Qt 5.15.X for ARM64 in GitHub Action. @richardssam 

Let us know if you managed to do it and if that runs well with Open RV. I haven't actually tested the binaries coming out of the GitHub Action for MacOS ARM64.

**Comment by richardssam** [comment]
These instructions are terrific. 

The instructions about checking openssl@1.1 were extremely useful, since the brew install --quiet --formula openssl@1.1 failed to install 1.1, and I had to run it again. It did give a warning that 1.1 is no longer supported. (it sounds like it will fail fairly soon).

Also, it seems like the qt config is not putting the final qt install in myQt/Qt/5.15.15/clang_64 but just myQt, should ----prefix="../myQt/Qt/5.15.15/clang_64"

I also needed to set DEVELOPER_DIR
export DEVELOPER_DIR=/Applications/XCode/14.3.1/Xcode.app/Contents/Developer
to use 14.3.1 since OSX had annoyingly upgraded my xcode.

But I have made it all the way to a built RV! Thanks so much.
